salt/modules/yumpkg.py

```diff
@@ -245,14 +245,27 @@ def _get_branch_option(**kwargs):
     Returns a list of '--branch' option to be used in the yum command,
     based on the kwargs. This feature requires 'branch' plugin for YUM.
     '''
-    branch = kwargs.get('branch', '')
+    branch = kwargs.pop('branch', '')
     ret = []
     if branch:
         log.info('Adding branch \'%s\'', branch)
         ret.append('--branch=\'{0}\''.format(branch))
     return ret
 
 
+def _get_extra_options(**kwargs):
+    '''
+    Returns list of extra options for yum
+    '''
+    ret = []
+    for key, value in kwargs.items():
```

Comment:
Should be using ``six.iteritems(kwargs)`` here. This will use ``.iteritems()`` on PY2 and ``.items()`` on PY3.